samples/Client/DaprClient/Program.cs

```diff
@@ -209,6 +212,34 @@ internal static async Task InvokeBalanceServiceOperationAsync(DaprClient client)
             Console.WriteLine($"Received balance {res.Balance}");
         }
 
+        internal static async Task InvokeThrowExceptionOperationAsync(DaprClient client)
+        {
+            Console.WriteLine("Invoking ThrowException");
+            var data = new { id = "17", amount = (decimal)10, };
+
+            HTTPExtension httpExtension = new HTTPExtension()
+            {
+                Verb = HTTPVerb.Post
+            };
+
+            try
+            {
+                // Invokes a POST method named "Withdraw" that takes input of type "Transaction" as define in the RoutingSample.            
+                await client.InvokeMethodAsync<object>("controller", "throwException", data, httpExtension);
+            }
+            catch (Grpc.Core.RpcException ex)
+            {
+                var entry = ex.Trailers.Get("grpc-status-details-bin");
+                var status = Google.Rpc.Status.Parser.ParseFrom(entry.ValueBytes);
+                Console.WriteLine("Grpc Exception Message: " + status.Message);
+                Console.WriteLine("Grpc Statuscode: " + status.Code);
+                var details = status.Details[0];
```

Comment:
There is a chance where `status.Details` have no item or multiple items. 
We need to iterate all Details and check if it is ErrorInfo protobuf or not before unpacking data to ErrorInfo. 

If possible I would create new exception class or extend RpcException to have [extension methods](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods) to get error method and status code.

For example, we can add the below four methods to RpcException using [extension methods](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods).

1. GetMessage -> status.message
2. GetCode -> status.Code
3. GetHTTPErrorMessage -> rpcError.Metadata["http.error_message"]
4. GetHTTPErrorCode -> rpcError.Metadata["http.code"]

So user code can call these methods simply.

Reply:
The goal for this PR isn't to design new APIs or add features, just to be an example of what's possible right now.

We're going to write a prososal for that separately.

Reply:
@rynowak I think we should fix runtime as well. The other gRPC APIs doesn't handle error correctly - https://github.com/dapr/dapr/issues/2067 We need to discuss about this issue as well.

Reply:
I fixed the part for iterating over the Details